priam/src/main/java/com/netflix/priam/backup/IncrementalBackup.java

```diff
@@ -116,11 +120,15 @@ protected void processColumnFamily(String keyspace, String columnFamily, File ba
                 backupRestoreConfig.enableV2Backups() ? BackupFileType.SST_V2 : BackupFileType.SST;
 
         // upload SSTables and components
-        upload(backupDir, fileType, config.enableAsyncIncremental(), true);
+        uploadAndDeleteAllFiles(backupDir, fileType, config.enableAsyncIncremental());
 
         // Next, upload secondary indexes
+        fileType = BackupFileType.SECONDARY_INDEX_V2;
+        ImmutableList<ListenableFuture<AbstractBackupPath>> futures;
         for (File dir : getSecondaryIndexDirectories(backupDir, columnFamily)) {
-            upload(dir, BackupFileType.SECONDARY_INDEX_V2, config.enableAsyncIncremental(), true);
+            futures = uploadAndDeleteAllFiles(dir, fileType, config.enableAsyncIncremental());
+            Futures.whenAllComplete(futures)
+                    .call(new DirectoryDeleter(dir), MoreExecutors.directExecutor());
```

Comment:
Curious on the choice of explicit class instantiation over using a lambda
 Futures.whenAllComplete(futures)
                    .call(() -> {
                        if (FileUtils.sizeOfDirectory(dir) == 0) FileUtils.deleteQuietly(dir);
                        return null;
                    }, MoreExecutors.directExecutor());

Reply:
Done.

Admittedly, I believe I tried that at first but couldn't figure it out. What you have works, however.